website/docs/script/sudt-script.mdx

```diff
@@ -0,0 +1,383 @@
+
+### Collecting UDT Amount
+
+In the sudt Script, the functions `collect_inputs_amount` and `collect_outputs_amount` play a crucial role in gathering the total amounts of UDTs from the respective input and output cells.
+
+Here’s how they work:
+
+Both functions iterate through the relevant cells and collect the `cell_data` to calculate the total UDT amount. This is done using the following syscalls:
+
+- **`Source::GroupInput`**: This ensures that only the cells with the same script as the current running script are iterating
+- **`Source::GroupOutput`**: Similarly, this syscall only the cells with the same script as the current running script are iterating
+
+By using these specific [sources](/docs/script/syscalls-for-script#source), the script avoids issues related to different UDT cell types, ensuring that only the appropriate cells are processed. This is particularly important in a blockchain environment where multiple UDTs may exist, and we want to ensure that the calculations are accurate and relevant to the current UDT Script.
+
+Here’s a brief look at how these functions are implemented:
+
+```rust
+fn collect_inputs_amount() -> Result<u128, Error> {
+    let mut buf = [0u8; UDT_AMOUNT_LEN];
+
+    let udt_list = QueryIter::new(load_cell_data, Source::GroupInput)
+        .map(|data| {
+            if data.len() >= UDT_AMOUNT_LEN {
+                buf.copy_from_slice(&data);
+                Ok(u128::from_le_bytes(buf))
+            } else {
+                Err(Error::AmountEncoding)
+            }
+        })
+        .collect::<Result<Vec<_>, Error>>()?;
+    Ok(udt_list.into_iter().sum::<u128>())
+}
+
+fn collect_outputs_amount() -> Result<u128, Error> {
+    let mut buf = [0u8; UDT_AMOUNT_LEN];
+
+    let udt_list = QueryIter::new(load_cell_data, Source::GroupOutput)
+        .map(|data| {
+            if data.len() >= UDT_AMOUNT_LEN {
+                buf.copy_from_slice(&data);
+                Ok(u128::from_le_bytes(buf))
+            } else {
+                Err(Error::AmountEncoding)
+            }
+        })
+        .collect::<Result<Vec<_>, Error>>()?;
+    Ok(udt_list.into_iter().sum::<u128>())
+}
+```
+
+In summary, these functions are essential for accurately calculating the total UDT amounts involved in the transaction while ensuring that only the relevant cells are considered, thus maintaining the integrity of the UDT transfer process.
+
+Full code: [contracts/sudt/src/main.rs](https://github.com/nervosnetwork/docs.nervos.org/tree/develop/examples/sudt-script/contracts/sudt/src/main.rs)
+
+### Writing Test For UDT Script
+
+Testing is a crucial part of developing any smart contract, including the sudt script.
+The `test_transfer_sudt` test case is designed to simulate a transfer of UDTs from one address to another.
+It sets up the necessary environment, including creating input and output cells, and then invokes the sudt Script to perform the transfer.
```

Comment:
It sets up the necessary environment, including creating input and output Cells, then invoking the sUDT Script to perform the transfer.